openedx/core/djangoapps/content_tagging/rules.py

```diff
@@ -274,6 +274,31 @@ def can_view_object_tag_objectid(user: UserType, object_id: str) -> bool:
     return bool(object_org) and (is_org_admin(user, object_org) or is_org_user(user, object_org))
 
 
+@rules.predicate
+def can_change_object_tag(
+    user: UserType, perm_obj: oel_tagging.ObjectTagPermissionItem | None = None
+) -> bool:
+    """
+    Returns True if the given user may change object tags with the given taxonomy + object_id.
+
+    Adds additional checks to ensure the taxonomy is available for use with the object_id's org.
+    """
+    if oel_tagging.can_change_object_tag(user, perm_obj):
+        if perm_obj and perm_obj.taxonomy and perm_obj.object_id:
+            is_all_org, taxonomy_orgs = TaxonomyOrg.get_organizations(perm_obj.taxonomy)
+            if not is_all_org:
+                # object_id is valid, else it would have thrown in can_change_object_tag_objectid
+                context_key = get_context_key_from_key_string(perm_obj.object_id)
+                assert context_key.org
```

Comment:
Similar to the question I had on https://github.com/openedx/edx-platform/pull/34200#discussion_r1489585528 , if we should handle the `AssertionError` and return `False` or does the `@rules.predicate` or elsewhere handle it?

Reply:
Agreed, I'll handle it and return False.